webExtensionApis/study/src/studyUtils.js

```diff
@@ -453,20 +304,6 @@ class StudyUtils {
     TelemetryEnvironment.setExperimentInactive(info.studyName);
   }
 
-  /**
-   * Uninstalls the shield study addon, given its addon id.
-   * @param {string} id - the addon id
-   * @returns {void}
-   */
-  uninstall(id) {
```

Comment:
What is the game plan here? Move uninstallation into webextension and https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/management/uninstall ?
I think it will be safer and easier for utils to provide a way to uninstall itself (probably via a timer) but we can of course test other ways of handling it.

Reply:
I claim that uninstalls should *only* happen these ways:
1.  User uninstalls addon directly (which shield/feature needs to react to)
2. User changes permission (which shield informs the study about)
3.  Normandy uninstalls
4.  User hits a 'feature defined' ending (expire, uninstall, good or bad ending), which the addon and shield need to react to.

I claim that *all of these* ssau should NOT be the one to uninstall directly, because the addon should not be *surprised* by the addon uninstalling.

Reply:
I agree about no surprise uninstall side effects. The follow-up question then is how the add-on should make the uninstallation effective. `browser.study.uninstall()` or 
```
setTimeout(() => {
  browser.management.uninstall(
    browser.runtime.id
  )
}, 1000);
```

I claim `browser.study.uninstall()` is cleaner and safer for the add-on developer.

Reply:
Just saw that https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/management/uninstall is not supported by Firefox yet, so we will need to have `browser.study.uninstall()` until further notice.